### PR TITLE
Renaming the response metadata key which was misrepresenting

### LINKED ISSUE #7144: Add `numConsumingSegmentsProcessed` and `numConsumingSegmentsMatched` into the query response metadata
Currently we have the following metadata in the query response:
- numSegmentsQueried: number of segments queried by the broker
- numSegmentsProcessed: number of segments processed after the server-side segment pruning
- numSegmentsMatched: number of segments that has records matched
- numConsumingSegmentsQueried: number of consuming segments queried by the broker

We want to also add `numConsumingSegmentsProcessed` and `numConsumingSegmentsMatched` to the query response metadata.

Note: In the `DataTable.MetadataKeys`, we mistakenly use `NUM_CONSUMING_SEGMENTS_PROCESSED` to represent the `numConsumingSegmentsQueried`

### PR DESCRIPTION
**Pull request by ravishankar15**
Partial Fix #7144

This PR will handle the naming mismatch mentioned in the issue description. Can we merge this first I will raise a separate PR in the following week to add the extra meta data values ?

I tried doing in the same PR it was so confusing 😕

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ enum MetadataKey {
     NUM_SEGMENTS_QUERIED("numSegmentsQueried", MetadataValueType.INT),
     NUM_SEGMENTS_PROCESSED("numSegmentsProcessed", MetadataValueType.INT),
     NUM_SEGMENTS_MATCHED("numSegmentsMatched", MetadataValueType.INT),
-    NUM_CONSUMING_SEGMENTS_PROCESSED("numConsumingSegmentsProcessed", MetadataValueType.INT),
```
Comment (siddharthteotia) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java:106]: Curious if this will be backward incompatible ?

Comment (ravishankar15) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java:106]: @siddharthteotia 
In the API response we where showing numConsumingSegmentsQueried which was computed using the key NUM_CONSUMING_SEGMENTS_PROCESSED. As part of this PR we are correcting **only the variable names which was incorrectly used not the API response** so as per my understanding there should not be any compatibility issue related to the same.

**Review by Jackie-Jiang — APPROVED** [review]
@siddharthteotia For data table v2, we might temporarily lose this stats because we use actual name to pass the metadata. For data table v3 (default), it is okay because we use ordinal to pass the metadata. IMO it is okay because by default we are using data table v3

**Comment by siddharthteotia** [comment]
> @siddharthteotia For data table v2, we might temporarily lose this stats because we use actual name to pass the metadata. For data table v3 (default), it is okay because we use ordinal to pass the metadata. IMO it is okay because by default we are using data table v3

yes that makes sense.

**Review by siddharthteotia — APPROVED** [review]

**Review by jackjlli — APPROVED** [review]

**Pull request merged**